topics/lmms/rqtl2/gn-rqtl-design-implementation.gmi

```diff
@@ -0,0 +1,118 @@
+# RQTL Implementation for GeneNetwork Design Proposal
+
+## Tags
+
+* Assigned: alexm,
+* Keywords: RQTL, GeneNetwork2, Design
+* Type: Enhancements,
+* Status: In Progress
+
+
+
+## Description
+
+This document outlines the design proposal for the re-implementation of the RQTL feature in GeneNetwork providing also  a console view to track the external process.
+
+### Problem Definition
+The current RQTL implementation requires enhancements. The core functionality remains the same: making API calls from GeneNetwork 3 (GN3). The system needs a cleaner architecture for future use, emphasizing improved error handling and a clear separation of concerns between GeneNetwork 2 (GN2) and GN3. This will involve eliminating file transfers between GN2 and GN3. Additionally, a console should be provided to users for tracking the progress of ongoing tasks.
+
+## High-Level Design
+This is divided into two major components:
+* RQTL Api
+
+* Monitoring system for the rqtl script
+
+### RQTL Api 
+This component serves as the entry point to the gn3 rqtl system and can be subdivided into
+data validation and preprocessing stage, data computation stage, output processing stage.
+In the component the task mainly   involves rewriting the current API to replace file transfers between GN2 and GN3 with JSON data We will also implement tests and enhance error handling.
+
+
+**Subtasks:**
```

Comment:
I'd suggest we have the subtasks be related to the discrete steps we've subdivided the RQTL processing.

Also, I think we're missing details for how we'll determine the progress of tasks. My suggesion would be:

```
The processing will be handled as a linked-list, where after each node, we can update the progress e.g.

data_validation_1 (20%)-> data_validation_2 (40%)-> preprocessing (60%)-> data computation (80%)->  output_processing(100%)
```

Reply:
this commit addresses this: https://github.com/genenetwork/gn-gemtext-threads/pull/46/commits/39caa87144e9b0dcab354d0043ccaed3d7c52e7e